internal/listener/listener.go

```diff
@@ -176,6 +185,12 @@ func (l *Listener) checkDebugPassword(w http.ResponseWriter, r *http.Request) bo
 }
 
 func (l *Listener) DumpConfig(w http.ResponseWriter, r *http.Request) {
+	if r.Method != http.MethodGet {
+		w.WriteHeader(http.StatusMethodNotAllowed)
+		_, _ = fmt.Fprintln(w, "Bad request method")
+		return
+	}
```

Comment:
I'd say something like "GET required" as it not only says you're doing something wrong but also what you should do instead.